src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -82,18 +86,14 @@ def fetch(self, sql) -> Iterator[any]:
         logger.debug(f"[api][fetch] {sql}")
         return self._sql.execute_fetch_all(self._warehouse_id, sql)
 
-    def save_table(self, full_name: str, rows: list[any], mode="append"):
+    def save_table(self, full_name: str, rows: list[any], klass: type, mode="append"):
         if mode == "overwrite":
             msg = "Overwrite mode is not yet supported"
             raise NotImplementedError(msg)
         rows = self._filter_none_rows(rows, full_name)
+        self.create_table(full_name, klass)
         if len(rows) == 0:
             return
-
-        klass = rows[0].__class__
```

Comment:
Previously we got the klass by looking at the first record, now we add it as the explicit argument. Do we really want to add explicit argument?